change-streams/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func ChangeStream(waitGroup sync.WaitGroup, stream *mongo.ChangeStream) {
```

Comment:
I'd change this to a private function and call it something like `iterateChangeStream`.